OpenSim/Common/Component.cpp

```diff
@@ -319,6 +331,11 @@ void Component::disconnect()
 
 void Component::addToSystem(SimTK::MultibodySystem& system) const
 {
+    // If being asked to be added to the same System that it is already
+    // a part, there is nothing to be done.
+    if (hasSystem() && (&getSystem() == &system)) {
+        return;
+    }
```

Comment:
Seems okay to me but I'm curious. What caused this to be necessary? Are you sure that silently passing is the correct thing to do, instead of throwing an exception?